content/guides/destructuring.adoc

```diff
@@ -200,6 +220,34 @@ This vector is comprised of nested vectors that we can access directly.
 ;= "Line from ( 5 , 10 ) to ( 10 , 20 )"
 ----
 
+This can become arbitrarily complex:
+
+[source,clojure]
+----
+(def my-struct '[ 1 2 ( 3 4 ) 5 [ ( 6 7 ) [ ( 8 ) 9 ] ] 10 ])
+----
+
+[source,clojure]
+----
+(let [ [a b c d [ [ e f ] [ g h ] ] i ]  my-struct ]
+  (apply prn [a b c d e f g h i ] ))
+;= 1 2 (3 4) 5 6 7 (8) 9 10
+
+(let [ [a b c d [ e f [ g h ] ] i ]  my-struct ]
+  (apply prn [a b c d e f g h i ] ))
+;= 1 2 (3 4) 5 (6 7) [(8) 9] nil nil 10
+----
+
+Of ourse, if the left-hand expression does not match the structure of the value, the destructuring operation will fail:
```

Comment:
I see no point in adding this example - let's focus on how to use it, not break it.

Reply:
Ok